server/lib/nl/counters.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+from typing import Dict
+
+
+#
+# A helper class to track info / error / timing counters.
+#
+class Counters:
+
+  def __init__(self):
+    self._warn: Dict = {}
+    self._info: Dict = {}
+    self._timing: Dict = {}
+
+  #
+  # Adds/updates an info counter.
+  #
+  # For a given counter, caller should always pass the same type
```

Comment:
if there are two different arg types for `value` then why not support two different functions? Or at least have two different versions of _update(), one for the numeric case and one for the list?

Reply:
This is called from a bunch of different places, and I was striving for something that is very low barrier to call (i.e., just know whether its info or error and pass "anything" to it really) :)